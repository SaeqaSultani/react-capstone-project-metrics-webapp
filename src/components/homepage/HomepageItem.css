.container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  column-gap: 2rem;
  justify-content: center;
  align-content: center;
  margin: 20px;
}

.item {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 400px;
  width: 340px;
  background-color: rgba(249, 228, 241);
  border-radius: 10px;
  margin-top: 30px;
  margin-bottom: 30px;
}

.img {
  height: 300px;
  width: 340px;
  border-radius: 10px 10px 0 0;
}

.title {
  font-weight: bold;
  font-family: Poppins, sans-serif;
  font-size: 35px;
  padding: 15px;
}
