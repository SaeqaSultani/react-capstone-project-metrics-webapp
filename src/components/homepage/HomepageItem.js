import React from 'react';
import PropTypes from 'prop-types';
import './HomepageItem.css';

const HomepageItem = (props) => {
  const { title } = props;
  HomepageItem.propTypes = {
    title: PropTypes.string.isRequired,
  };

  return (
    <div className="container">
      <div className="item">
        <h1 className="title">{title}</h1>
      </div>
    </div>
  );
};

export default HomepageItem;
