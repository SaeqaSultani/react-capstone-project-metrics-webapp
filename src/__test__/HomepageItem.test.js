import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';
import HomepageItem from '../components/homepage/HomepageItem';

it('Check any changes to the component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <HomepageItem />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render a div container', () => {
  render(
    <Provider store={store}>
      <HomepageItem  title="title"/>
    </Provider>,
  );

  const page = screen.getByTestId('homepageitem');

  expect(page).toBeInTheDocument();
});
