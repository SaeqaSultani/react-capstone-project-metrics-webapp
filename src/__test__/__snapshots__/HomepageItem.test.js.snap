// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check any changes to the component 1`] = `
<div
  className="container"
  data-testid="homepageitem"
>
  <div
    className="item"
  >
    <img
      alt="company"
      className="icon"
      src="company.png"
    />
    <h1
      className="title"
    />
  </div>
</div>
`;
